api/repo.go

```diff
@@ -372,6 +372,146 @@ func GetRepos(c *gin.Context) {
 }
 
+// swagger:operation GET /api/v1/{org} repos GetOrgRepos
+//
+// Get all repos in the configured backend
+//
+// ---
+// produces:
+// - application/json
+// security:
+//   - ApiKeyAuth: []
+// parameters:
+// - in: path
+//   name: org
+//   description: Name of the org
+//   required: true
+//   type: string
+// - in: query
+//   name: page
+//   description: The page of results to retrieve
+//   type: integer
+//   default: 1
+// - in: query
+//   name: per_page
+//   description: How many results per page to return
+//   type: integer
+//   maximum: 100
+//   default: 10
+// responses:
+//   '200':
+//     description: Successfully retrieved the repo
+//     schema:
+//       type: array
+//       items:
+//         "$ref": "#/definitions/Repo"
+//     headers:
+//       X-Total-Count:
+//         description: Total number of results
+//         type: integer
+//       Link:
+//         description: see https://tools.ietf.org/html/rfc5988
+//         type: string
+//   '400':
+//     description: Unable to retrieve the org
+//     schema:
+//       "$ref": "#/definitions/Error"
+//   '500':
+//     description: Unable to retrieve the org
+//     schema:
+//       "$ref": "#/definitions/Error"
+
+// GetOrgRepos represents the API handler to capture a list
+// of repos for a org from the configured backend.
+
+func GetOrgRepos(c *gin.Context) {
+	// capture middleware values
+	u := user.Retrieve(c)
+	org := c.Param("org")
+	logrus.Infof("Reading repos for org %s", org)
+
+	// capture page query parameter if present
+	page, err := strconv.Atoi(c.DefaultQuery("page", "1"))
+	if err != nil {
+		retErr := fmt.Errorf("unable to convert page query parameter for user %s: %w", u.GetName(), err)
+
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+
+	// capture per_page query parameter if present
+	perPage, err := strconv.Atoi(c.DefaultQuery("per_page", "10"))
+	if err != nil {
+		// nolint: lll // ignore long line length due to error message
+		retErr := fmt.Errorf("unable to convert per_page query parameter for user %s: %w", u.GetName(), err)
+
+		util.HandleError(c, http.StatusBadRequest, retErr)
+
+		return
+	}
+
+	// ensure per_page isn't above or below allowed values
+	//
+	// nolint: gomnd // ignore magic number
+	perPage = util.MaxInt(1, util.MinInt(100, perPage))
+
+	allRepos, err := database.FromContext(c).GetOrgPrivateRepoList(org)
+	if err != nil {
+		logrus.Errorf("unable to get private repos for org %s : %s", org, err)
+	}
+
+	var excludeList []string
+
+	for _, rr := range allRepos {
+		perm, err := source.FromContext(c).RepoAccess(u, rr.GetOrg(), rr.GetName())
+		if err != nil {
+			logrus.Errorf("unable to get user %s access level for repo %s", u.GetName(), rr.GetFullName())
+		}
+
+		switch perm {
+		case "admin", "write", "read":
+			continue
+		default:
+			excludeList = append(excludeList, rr.GetName())
+		}
+	}
+	if len(excludeList) == 0 {
+		excludeList = append(excludeList, "")
+	}
```

Comment:
do we actually need an blank entry? could we instead of `var excludeList []string` do `excludeList := []string{}` and avoid this?

Reply:
I did that initially, and the query didn't return any results. The empty string will just add an empty string in the where clause. 
```sql
SELECT ...
WHERE name NOT IN ("")
```
And there will never be a repo with an empty string as a name.

Reply:
OK thanks, that's what I feared. Can we add a comment above the code block that adds the empty entry to call that out more?